tools/missing-test-detector/detector.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"strings"
+)
+
+type MissingTestInfo struct {
+	UntestedFields []string
+	Tests          []string
+}
+
+type FieldSet map[string]struct{}
+
+// Detect missing tests for the given resource changes map in the given slice of tests.
+// Return a map of resource names to missing test info about that resource.
+func detectMissingTests(changedFields map[string]FieldCoverage, allTests []*Test) map[string]*MissingTestInfo {
```

Comment:
It would be great to separate the concerns here a little more - that is, for the input of this function to be a list of fields that have changed, and then have this function construct a separate object that tracks whether those fields are covered or not. This would make it easier for us in the future to reuse the schema diffing logic for things like breaking change detection, or any other similar functionality.

Reply:
That said - no need to change this for this PR. We should track a separate project to bring these tools in alignment later, probably when we have a third example of a tool that needs to detect schema changes.